src/components/Characters/Characters.js

```diff
@@ -17,38 +36,175 @@ const Characters = () => {
     }, [info, url,search, results])
   
     useEffect(()=>{                     //combined url + result in input for search filter purpose
-        fetch(`${url}${search}`, {
+        fetch(`${url}?page=${page}&name=${search}&status=${currentStatus}&gender=${currentGender}&species=${currentSpecies}`, {
           method: 'GET'
         }).then((response) => {
-          return response.json()
+            if (response.status === 404) {
+                setError(true)
+                throw new Error('Smth wrong')
+            }
+        return response.json()
         }).then((result) => {
             setInfo(result.info)
             setResults(result.results)
+
+            
         }).catch((error) => {
+            setPage(1)
             console.log(error)
         })
-    }, [search])                        //Serach as a main dependency for this 
-  
+    }, [search, page, url, currentStatus, currentGender, currentSpecies, error])                       //Serach as a main dependency for this 
+
+    const HandlerBefore = (event) => {
```

Comment:
HandlerBefore has 3 problems in the same name
 - should start from the lower case latter
 - the first word in the name should be a verb since function = action
 - before is too general to name a handler, any other name could be better, ie handleButtonClick or handleImagesLoad, etc

Reply:
Sure. I should include more sense in naming functions/varibales or smth ellse